src/main/java/hudson/plugins/mercurial/MercurialSCM.java

```diff
@@ -881,7 +881,11 @@ private void clone(Run<?, ?> build, Launcher launcher, FilePath repository, Node
     }
 
     @Override
-    public void buildEnvVars(AbstractBuild<?,?> build, Map<String, String> env) {
+    public void buildEnvVars(AbstractBuild<?,?> build, Map<String,String> env) {
+        buildEnvironment(build, env);
+    }
+
+    public void buildEnvironment(Run<?,?> build, Map<String, String> env) {
```

Comment:
Same as for SVN plugin. It is better to reference the incoming method in the Core API.